lib/ucx_ucc/ucx_ucc_cert_manager.ex

```diff
@@ -0,0 +1,96 @@
+# Copyright (C) E-MetroTel, 2018 - All Rights Reserved
+# This software contains material which is proprietary and confidential
+# to E-MetroTel and is made available solely pursuant to the terms of
+# a written license agreement with E-MetroTel.
+
+
+
+defmodule UcxUcc.CertManager do
+  @moduledoc """
+
+  This module gets SSL certificates file path from ssl configuration file for use in UcxUcc application.
+
+  General functions to manage ssl certificate files on a UCx.
+  """
+
+  require Logger
+
+  @ssl_conf_file "/etc/httpd/conf.d/ssl.conf"
+  @ssl_cert "SSLCertificateFile"
+  @ssl_cert_key "SSLCertificateKeyFile"
+  @ssl_ca_cert "SSLCACertificateFile"
+
+  @doc """
+  Parses the ssl configuration file and returns list of file names.
+
+  Returns the `certfile`, `cacertfile`, and `keyfile` file names.
+  If the `cacertfile` is not found, the `certfile` will be returned.
+
+  ## Examples
+
+      get_cert_file        # uses the default #{@ssl_conf_file} file
+
+      get_cert_file(:mscs) # gets the config filename from the config
+                           # using the app key give (`:mscs`)
+
+      get_cert_file("/etc/ssl.conf")  # uses config file name given
+
+  """
+  def get_cert_info(name \\ @ssl_conf_file)
+  def get_cert_info(app) when is_atom(app) do
+    Application.get_env(app, :ssl_conf_file, @ssl_conf_file)
+    |> get_cert_info
+  end
+  def get_cert_info(ssl_conf_file) do
+    File.stream!(ssl_conf_file, [], :line)
+    |> Enum.reduce([], fn(line, acc) ->
+      line = Regex.replace ~r/#.*/, line, ""
+      cond do
```

Comment:
Few things in this block. 

1. `find_key` already trims the string. So no need to do it a second time.
2. Code guidelines

```
cond do
  value = find_key(@ssl_cert, line) ->
    acc
    |> Keyword.put(:certfile, value)
    |> Keyword.put_new(:cacertfile, value)
  value = find_key(@ssl_cert_key, line) ->
    Keyword.put(acc, :keyfile, value)
  value = find_key(@ssl_ca_cert, line) ->
    Keyword.put(acc, :cacertfile, value))
  true -> acc
end
```

Reply:
Updated code.